platform/darwin/docs/guides/Data-Driven Styling.md.ejs

```diff
@@ -0,0 +1,122 @@
+
+`MGLStyleFunction`
+
+There are three subclasses of `MGLStyleFunction`:
+
+* `MGLCameraStyleFunction` - For a style value that changes with zoom level. For example, you can make the radius of a circle to increase based on zoom level. In iOS SDK v3.4, this was a `MGLStyleFunction`.
+* `MGLSourceStyleFunction` - For a style value that changes with the attributes of a feature. You can adjust the radius of a circle based on the magnitude of an earthquake, for example.
+* `MGLCompositeStyleFunction` - For a style value that changes with both zoom level and attribute values. For example, you can add a circle layer where each circle has a radius based on both zoom level and the magnitude of an earthquake.
+
+The documentation for individual style properties will note which style functions are enabled for that property.
+
+### Stops
+
+Stops are key-value pairs that that determine a style value. With a `MGLCameraSourceFunction` stop, you can use a dictionary with a zoom level for a key and a `MGLStyleValue` for the value. For example, you can use a stops dictionary with zoom levels 0, 10, and 20 as keys, and yellow, orange, and red as the values. A `MGLSourceStyleFunction` uses the relevant attribute value as the key.
+
+### Interpolation Mode
+
+The effect a key has on the style value is determined by the interpolation mode. There are four interpolation modes that can be used with a source style function: exponential, interval, categorical, and identity. You can also use exponential and interval interpolation modes with a camera style function.
+
+#### Exponential (or Linear)
+
+`MGLInterpolationModelExponential` creates a linear or exponential effect based on the values. The key value is the starting point for interpolation, and the style value is based on where an attribute value falls between two keys. By default, `MGLInterpolationModeExponential` uses an interpolation base of 1. This causes a linear relationship between the style value and stop key.
+
+The stops dictionary below, for example, shows colors that continuously shift from yellow to orange to red to blue to white based on the attribute value.
+
+``` swift
+    let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson")
+    let symbolSource = MGLSource(identifier: "source")
+    let symbolLayer = MGLSymbolStyleLayer(identifier: "place-city-sm", source: symbolSource)
+
+    let source = MGLShapeSource(identifier: "earthquakes", url: url, options: nil)
+    style.addSource(source)
+
+    let stops = [0 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.yellow),
+                 2.5 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.orange),
+                 5: MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.red),
+                 7.5 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.blue),
+                 10 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.white)]
```

Comment:
Remove the spaces before the colons.

![screen shot 2017-03-17 at 4 59 55 pm](https://cloud.githubusercontent.com/assets/1198851/24062778/39964988-0b33-11e7-8345-65fd043a7c90.png)